docs/developers/fuelmint.md

```diff
@@ -0,0 +1,163 @@
+
+### Celestia
+
+Celestia is a modular blockchain network whose goal
+is to build a scalable data availability layer,
+enabling the next generation of scalable blockchain
+architectures - modular blockchains.
+
+You can learn more about Celestia in our
+[Concepts](../../concepts/how-celestia-works/introduction)
+and [Build Modular](../build-modular) pages.
+
+### Fuelmint
+
+Fuelmint is a Sovereign FuelVM rollups on Celestia using
+Rollmint and ABCI. You can read the announcement blog post
+[here](https://blog.celestia.org).
+
+In this tutorial, we will show you how to setup Fuelmint, so
+you can test out the advantages of using Sway while having a sovereign
+rollup on Celestia.
+
+Let's get started!
+
+## Installation
+
+### Dependencies
+
+You would need to install the following dependencies on your machine
+to be able to setup Fuelmint.
+
+* [Install Rust and Cargo](https://doc.rust-lang.org/cargo/getting-started/installation.html)
+* [Install Docker](https://docs.docker.com/engine/install/ubuntu/)
+* [Install System Environment Setup for Linux, including Golang](../../nodes/environment)
+
+### Setup Fuelmint
+
+Fuelmint Repo is [here](https://github.com/Ferret-san/fuelmint/tree/tx_pool_experiment)
+
+First step is to clone the Fuelmint repo:
+
+```bash
+git clone https://github.com/Ferret-san/fuelmint
+```
+
+Then go to the docker directory:
+
+```bash
+cd fuelmint/local-da
+```
+
+You can start by running a Docker Compose
+setup with a local DA network
+
+```bash
+docker compose -f ./docker/test-docker-compose.yml up
+```
+
+In a separate terminal session, you must
+build the Fuelmint binary with `cargo`:
+
+```bash
+cd ..
+cargo run --bin fuelmint
+```
+
+In another terminal session, you will need
+to build the Rollkit Node with Golang:
+
+```bash
+cd rollkit-node
+go mod edit -replace github.com/cosmos/cosmos-sdk=github.com/celestiaorg/cosmos-sdk-rollmint@v0.46.7-rollmint-v0.5.0-no-fraud-proofs
+go mod edit --replace github.com/tendermint/tendermint=github.com/celestiaorg/tendermint@v0.34.22-0.20221013213714-8be9b54c8c21
+go mod tidy
+go mod download
+make install
+```
+
+Then run the following commands:
+
+<!-- markdownlint-disable MD013 -->
+```bash
+rm -rf /tmp/fuelmint/
+TMHOME="/tmp/fuelmint" tendermint init
+NAMESPACE_ID=$(echo $RANDOM | md5sum | head -c 16; echo;)
+./rollkit-node -config "/tmp/fuelmint/config/config.toml" -rollkit.namespace_id $NAMESPACE_ID -rollkit.da_start_height 1
+```
+<!-- markdownlint-enable MD013 -->
+
+## Deploy A Sway Smart Contract
+
+### Install Forc
+
+* Install Forc
+
+### Counter Smart Contract
+
+Find the Sway smart contracts [here](https://fuellabs.github.io/fuels-ts/QUICKSTART)
+
+Let's deploy a Sway smart contract for counter!
+
+```bash
+cd ..
+cd contract/
+forc build
+forc deploy --url localhost:4000 --unsigned
+```
+
+This generates the contract id
+
+Generate the front end with contract-id
+Get the wallet secret generated when you started fuelmint
```

Comment:
I plan to add a note when this happens, so user doesn't need to go back and find it after it is deep in terminal outputs

Reply:
is there a best practice for this?